gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -150,117 +201,300 @@ def test_unauthorized_user_cannot_list_tes_tasks(self):
 
     def test_happy_path_cancel_tes_tasks(self):
         """
         Verify that an authorized user can cancel a TES task.
         Expects: HTTP 200 and task status changes to 'Cancelled'.
         """
-        response = self.gen3_workflow.create_tes_task(
+        payload = {
+            "name": "Hello world",
+            "description": "Demonstrates the most basic echo task.",
+            "executors": [
+                {
+                    "image": "public.ecr.aws/docker/library/alpine:latest",
+                    "command": ["echo", "hello beautiful world!"],
+                }
+            ],
+            "tags": {"user": "bar"},
+        }
+        # Step 1: Create a TES task
+        task_info = self.gen3_workflow.create_tes_task(
+            request_body=payload,
             user=self.valid_user,
             expected_status=200,
         )
-        assert (
-            "id" in response
-        ), f"Response should contain a valid 'id' field. But found {response} instead"
-        task_id = response["id"]
+        task_id = task_info.get("id", None)
+        assert task_id, f"Expected 'id' in response, but got: {task_info}"
 
-        response = self.gen3_workflow.get_tes_task(
+        # Step 2: Retrieve the task to check its state
+        task_info = self.gen3_workflow.get_tes_task(
             task_id=task_id,
             user=self.valid_user,
             expected_status=200,
         )
+
         assert (
-            "state" in response
-        ), f"Response must have `state` field in it. But found {response} instead"
+            "state" in task_info
+        ), f"Response must have `state` field in it. But found {task_info} instead"
 
-        response = self.gen3_workflow.cancel_tes_task(
+        # Step 3: Cancel the TES task
+        self.gen3_workflow.cancel_tes_task(
             task_id=task_id,
             user=self.valid_user,
             expected_status=200,
         )
-        assert "state" in response and response["state"] in [
-            "CANCELING",
-            "CANCELED",
-        ], f"Task state should be 'CANCELING', 'CANCELED' after cancellation. But found {response} instead"
+
+        # Step 4: Verify the task is cancelled
+        task_info = self.gen3_workflow.get_tes_task(
+            task_id=task_id,
+            user=self.valid_user,
+            expected_status=200,
+        )
+
+        final_state = task_info.get("state", None)
+        valid_states = {"CANCELING", "CANCELED"}
+        assert (
+            final_state in valid_states
+        ), f"Task state should be one of {valid_states} after cancellation. But found {final_state} instead. Task Info: {task_info}"
+
+    def test_nextflow_workflow(self):
+        """
+        Test Case: Verify that a Nextflow workflow can be executed successfully.
+        """
+        expected_task_outputs = {
+            "extract_metadata (1)": {
```

Comment:
wondering if we could/should also access the command stdout and check it? for example one of your tasks should be outputing "Resolved workDir [...]"

Reply:
Can you elaborate a little on this one? I'll push a commit addressing the rest.

Reply:
in addition to the csv and png files, the outputs in the s3 bucket should also include .command.sh, .command.run, command.err, command.out, command.log and .exitcode

Reply:
I don't think checking the contents of `.command.run` is necessary, I was thinking about the output files - `.out` `.err` and `.exitcode` maybe